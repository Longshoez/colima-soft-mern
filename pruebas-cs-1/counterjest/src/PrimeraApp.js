import React from 'react';

const PrimeraApp = ({saludo, subtitulo = "Yo yo yo"}) => {
  return(
    <>
        <h2>{saludo}</h2>
        <h6>{subtitulo}</h6>
    </>
)};

export default PrimeraApp