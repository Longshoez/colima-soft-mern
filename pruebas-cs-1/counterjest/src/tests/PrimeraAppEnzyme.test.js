import React from 'react';
import {render} from '@testing-library/react'
import PrimeraApp from "../PrimeraApp";
import { shallow } from 'enzyme'

describe('Pruebas en el componente <PrimeraApp /> usando Enzyme', () => {
    test('should show the <PrimeraApp/> correctamente', () => {
      //Arrange
      const saludo = "Hi there Gabriel"
      
      //Act
      const wrapper = shallow(<PrimeraApp saludo={saludo}/>)
      expect(wrapper).toMatchSnapshot();

      //Assert
    });


    test('must show the subtitle sent by props', () => {
      //Arrange
      const saludo = "Hi there Gabriel"
      const subtitulo = "yo yo yo"
      
      //Act
      const wrapper = shallow(
        <PrimeraApp 
        saludo={saludo} 
        subtitulo={subtitulo}
        />
        )
        const subTexto = wrapper.find('h6').text()

      //Assert
      expect(subTexto).toBe(subTexto)

    });    
})