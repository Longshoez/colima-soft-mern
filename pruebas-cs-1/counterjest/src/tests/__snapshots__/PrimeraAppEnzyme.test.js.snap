// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el componente <PrimeraApp /> usando Enzyme should show the <PrimeraApp/> correctamente 1`] = `
<Fragment>
  <h2>
    Hi there Gabriel
  </h2>
  <h6>
    Yo yo yo
  </h6>
</Fragment>
`;
